azure-application-insights-spring-boot-starter/src/main/java/com/microsoft/applicationinsights/boot/ApplicationInsightsWebMvcAutoConfiguration.java

```diff
@@ -56,7 +56,9 @@ public FilterRegistrationBean webRequestTrackingFilterRegistrationBean(WebReques
 
     @Bean
     @ConditionalOnMissingBean
-    public WebRequestTrackingFilter webRequestTrackingFilter(@Value("${spring.application.name:application}") String applicationName) {
+    public WebRequestTrackingFilter webRequestTrackingFilter(@Value("${spring.application.name:application}") String applicationName,
+            // we have implicit dependency on configured telemetryConfiguration here
+            @SuppressWarnings("unused") TelemetryConfiguration telemetryConfiguration) {
```

Comment:
@gavlyukovskiy  Can you explain the need of this? I do understand in the changes above you made this conditional on web application which is auto configured after the default configuration. Is this what you are referring to and believe that TelemetryConfiguration would always be initialized ?

Reply:
`@AutoConfigureAfter` and `@AutoConfigureBefore` sets the order of creating bean definitions, but the beans itself are created in different order which controlled by dependency graph. I set explicit dependency on `TelemetryConfiguration` to force it to be created before `WebRequestTrackingFilter` to ensure that call `TelemetryConfiguration#getTelemetryModules` will return all modules.

Reply:
@gavlyukovskiy thank you very much. Also about the sampling do you have any idea on getting that in?

Reply:
Yes so if processor returns false then the telemetry is dropped and doesn't hit transmission channel. So processor controls if telemetry should be sent or not. Channel is responsible for ultimately pumping the telemetry and it has several characterstics which tell what is the end point what is the buffer size etc.

Reply:
And for what `TelemetrySampler` is used?

Reply:
Yes please drop it. It is not a recommended way to sample. We should mark it as deprecated also :-)

Reply:
What do you think about moving these types of telemetry to enum:
* Dependency
* Event
* Exception
* PageView
* Request
* Trace
? With that we can make autocomplete in properties.

Reply:
@gavlyukovskiy are you referring to converting this class into ENUMS inside core SDK or specifically in this module? In either cases I think that would be a major refactoring. Could you add more details on why do we need this? What benefits it provides and what would we be loosing here if we keep it to what it is currently. Further all these data types are autogenerated by a C# script I think. We might need to make changes there to do this. Also there might be upcoming change with Schema V 3.0 . Let's discuss more on this.

Reply:
@gavlyukovskiy I am okay with this unless this is something globally exposed. Autocompletion would be desirable but I would like to understand what could be the implications if this is globally exposed because we won't be able to remove this until next major version and making it public then people can take explicit dependency on this enum and if we want to make it public in any situation then it should not reside in this processor but rather should have it's home somewhere else where it is more easy to locate and used in other parts of the code base.